src/Tiqr/TiqrFactory.php

```diff
@@ -15,94 +18,63 @@
  * limitations under the License.
  */
 
-namespace App\Tiqr;
+namespace Surfnet\Tiqr\Tiqr;
 
-use App\Tiqr\Legacy\TiqrService;
-use App\Tiqr\Legacy\TiqrUserRepository;
 use Psr\Log\LoggerInterface;
-use Symfony\Component\DependencyInjection\ContainerInterface;
-use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Surfnet\Tiqr\Tiqr\Legacy\TiqrService;
+use Surfnet\Tiqr\Tiqr\Legacy\TiqrUserRepository;
+use Symfony\Component\HttpFoundation\RequestStack;
 use Tiqr_Service;
 use Tiqr_StateStorage;
 use Tiqr_UserSecretStorage;
 use Tiqr_UserStorage;
 
 class TiqrFactory
 {
-    private $configuration;
-    private $container;
-    private $session;
-    private $logger;
-    private $loaded = false;
-
-    public function __construct(
-        TiqrConfigurationInterface $configuration,
-        ContainerInterface $container,
-        SessionInterface $session,
-        LoggerInterface $logger
-    ) {
-        $this->configuration = $configuration;
-        $this->container = $container;
-        $this->session = $session;
-        $this->logger = $logger;
-    }
-
     // Created from services.yaml
-    public function createService(): TiqrService
-    {
-        $this->loadDependencies();
-        $options = $this->configuration->getTiqrOptions();
+    public static function createService(
+        TiqrConfigurationInterface $configuration,
+        RequestStack $requestStack,
+        LoggerInterface $logger,
+        string $appSecret
+    ): TiqrService {
+        $options = $configuration->getTiqrOptions();
 
         $storageType = "file";
-        $storageOptions = array();
+        $storageOptions = [];
 
         if (isset($options["statestorage"])) {
             $storageType = $options["statestorage"]["type"];
             $storageOptions = $options["statestorage"];
         }
 
         return new TiqrService(
-            new Tiqr_Service($this->logger, $options),
-            Tiqr_StateStorage::getStorage($storageType, $storageOptions, $this->logger),
-            $this->session,
-            $this->logger,
+            new Tiqr_Service($logger, $options),
```

Comment:
TiqrService and Tiqr_Service are confusing

Reply:
Yup, I'm fully aware of this mixup. We are using the Tiqr server lib (Tiqr_Service legacy code) and try to port it to the GSSP via the TiqrService.